### DISCUSSION TITLE
분산락 구현 - 동시성 문제해결

### DISCUSSION THREAD

**Discussion by waterfogSW** [Student-Center/weave-server#88] [개발 논의]
- 앞선 두 PR(#61, #86)에서 논의되었던 동시성 문제를 해결하기 위한 분산락 구현이 필요해요.

**Comment by waterfogSW** [comment]
어차피 현재 redis도 붙여서 운용중이고, 추후 scale out도 고려하면 redisson 쓰는게 좋을것 같은데 어떤가요🙄

코드에 적용할때는 AOP활용해서 어노테이션으로 Lock잡으면 좋을것 같아요!(https://helloworld.kurly.com/blog/distributed-redisson-lock/)

**Reply by wonjunYou** [reply]
사실 분산락에서 걸리는 점이 redis같은 추가 인프라가 생긴다는 점인데, 저희 서비스에선 이미 있으니 저도 redission이 제일 베스트일 것 같네요!!
